manifests/install.pp

```diff
@@ -0,0 +1,22 @@
+# @summary Install pulpcore packages
+#
+# Currently this is done with pip3. This should have the option to also install via RPMs.
+#
+# Names are placeholders while I determine what packages need to be pip installed.
+#
+# @example
+#   include pulpcore::install
+class pulpcore::install {
+
+  ensure_packages(['package1', 'package2', 'package3'], {
+    ensure   => present,
+    provider => 'pip3',
+    require  => [ Package['python3-pip'], ],
+  })
+
+  file { '/etc/httpd/conf.d/pulp-vhost.conf':
```

Comment:
We always use the [apache](https://github.com/puppetlabs/puppetlabs-apache) module. It provides [apache::vhost](https://github.com/puppetlabs/puppetlabs-apache/blob/master/REFERENCE.md#apachevhost). https://github.com/theforeman/puppet-pulp/blob/master/manifests/crane/apache.pp is probably a good minimal example though https://github.com/theforeman/puppet-foreman_proxy_content/blob/master/manifests/reverse_proxy.pp is probably closer to how you'll deploy this.

I'd recommend putting the apache configuration in its own file `manifests/apache.pp`.

Reply:
I think this as a first pass, especially as @wbclark is learning puppet at the same time is good for just getting things to work. We discussed then layering in the next level of enhancements as he goes.

I will add that we will more likely be tackling this via something similar to (https://github.com/theforeman/puppet-katello/blob/master/manifests/application.pp#L75-L77) given we will need to add reverse proxy statements for Pulp to the Foreman vhost definition since it controls 443.

Reply:
```puppet
$pulp_api_path = '/pulp/api'
$pulp_api_url = "http://${pulp::pulp_api_host}:%{pulp::pulp_api_port}/pulp/api"
$pulp_content_path = '/pulp/content'
$pulp_content_url = "http://${pulp::pulp_content_host}:%{pulp::pulp_content_port}/pulp/content"

include apache
apache::vhost { 'pulp':
  servername => $facts['fqdn'],
  port       => 80,
  docroot    => $pulp::pulp_webserver_static_dir,
  proxy_pass => [        
    {
      'path'         => $pulp_api_path,
      'url'          => $pulp_api_url,
      'reverse_urls' => [$pulp_api_path, $pulp_api_url],
    },
    {
      'path'         => $pulp_content_path,
      'url'          => $pulp_content_url,
      'reverse_urls' => [$pulp_content_path, $pulp_content_url],
    },
  ],
}

```

Reply:
Assume this is just for early testing purposes? @ekohl

Reply:
Bit of a lack of context, but I explained how other modules do this to @wbclark and I needed to write it down somewhere. Now it's a convenient reference. It's untested, but this should be the equivalent of the vhost file with the benefit that it can be used together with the rest of the installer modules. In the future it should also be easy to add SSL to it, but that's for a later session to explain :)